File: src/main/java/com/example/ecommerceb2b/entities/Pedido.java
package com.example.ecommerceb2b.entities;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Pedido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;
    public BigDecimal valorTotal;
    public LocalDateTime data;
    public BigDecimal desconto;
}

File: frontadmin/app/types/pedido.ts
export class Pedido {

    constructor(
        public id:number | null,
        public valorTotal:number,
        public data:string,
        public desconto:number,
        public status:string
    ){

    }
}

export interface PedidoFormProp{
    pedidoExistente?:Pedido
}
